### PR TITLE
la::p::Vector: split compress() and update_ghosts() calls into chunks

### PR DESCRIPTION
**Pull request by davydden**
I would say for now fixes https://github.com/dealii/dealii/issues/6015

### CODE REVIEW

**Comment by davydden** [comment]
@tjhei looks like `tjhei-alpha` got stuck

File: include/deal.II/lac/la_parallel_block_vector.templates.h [hunk 1]
```diff
@@ -241,15 +241,23 @@ namespace LinearAlgebra
...
-      // this saves repeated synchronizations. In order to avoid conflict with
-      // possible other ongoing communication requests (from
-      // LA::distributed::Vector that supports unfinished requests), add an
-      // arbitrary number 8273 to the communication tag
-      for (unsigned int block=0; block<this->n_blocks(); ++block)
-        this->block(block).compress_start(block + 8273, operation);
-      for (unsigned int block=0; block<this->n_blocks(); ++block)
-        this->block(block).compress_finish(operation);
+      const unsigned int n_chunks = this->n_blocks()/communication_block_size +
+                                    (this->n_blocks()%communication_block_size > 0);
```
Comment (kronbichler) [inline include/deal.II/lac/la_parallel_block_vector.templates.h]: we often write this as
```
(this->n_blocks()+communication_block_size-1)/communication_block_size
```

File: include/deal.II/lac/la_parallel_block_vector.h [hunk 2]
```diff
@@ -81,7 +81,12 @@ namespace LinearAlgebra
     class BlockVector : public BlockVectorBase<Vector<Number> >,
       public VectorSpaceVector<Number>
     {
+      /**
+       * The chunks size to split communication in update_ghost_values() and compress() calls.
```
Comment (kronbichler) [inline include/deal.II/lac/la_parallel_block_vector.h]: Can you wrap the comment line at 80 characters? Please also say that this value has been found viable in experimentation. (I think it is a very good choice.)

File: include/deal.II/lac/la_parallel_block_vector.templates.h [hunk 3]
```diff
@@ -258,13 +266,21 @@ namespace LinearAlgebra
...
     {
-      // In order to avoid conflict with possible other ongoing communication
-      // requests (from LA::distributed::Vector that supports unfinished
-      // requests), add an arbitrary number 9923 to the communication tag
-      for (unsigned int block=0; block<this->n_blocks(); ++block)
-        this->block(block).update_ghost_values_start(block + 9923);
-      for (unsigned int block=0; block<this->n_blocks(); ++block)
-        this->block(block).update_ghost_values_finish();
+      const unsigned int n_chunks = this->n_blocks()/communication_block_size +
+                                    (this->n_blocks()%communication_block_size > 0);
```
Comment (kronbichler) [inline include/deal.II/lac/la_parallel_block_vector.templates.h]: same as above

**Comment by davydden** [comment]
@kronbichler i updated the PR.

> Please also say that this value has been found viable in experimentation

Frankly, I have not tried other values like `50`. What I know from similar context is that limiting helps in general, but I did not check for a specific best value.

**Comment by kronbichler** [comment]
> Frankly, I have not tried other values like 50. What I know from similar context is that limiting helps in general, but I did not check for a specific best value.

Even without a careful experimentation (which is beyond the scope here I agree), we can give a heuristic argument that
- most common MPI implementations will get slow when too many messages/requests are outstanding,
- even when messages are small, say 1 kB only, we have collected enough data with 20 of those to cover typical infiniband latencies which are around a few microseconds. Sending 20 kB at a throughput of 5 GB/s takes 4 microseconds, so we should come into the bandwidth dominated regime then which is good enough.

**Comment by davydden** [comment]
@kronbichler do you want to push a docu update with this argument to my branch so that you get the credit for it 😄

**Comment by kronbichler** [comment]
Since you're at the branch, please do it yourself. It's our common project, so I don't really care - and I don't think I get too little credit out of it anyway. :smile:

**Comment by davydden** [comment]
@kronbichler i updated, please have a look.

**Comment by davydden** [comment]
@kronbichler since the last tester seems to hang, i guess we can merge this

**Comment by davydden** [comment]
@kronbichler for the record, this PR also solved an ugly MPI issue that my computations hanged on the cluster at some point. I suppose there was  some interference between those non-blocking send/recieve that somehow got triggered for specific size of the system in terms of DoFs and number of bocks.

**Pull request merged**